File: event-consumer/src/test/java/nl/haploid/octowight/kafka/KafkaConsumerFactoryTest.java
package nl.haploid.octowight.kafka;

import kafka.consumer.ConsumerConfig;
import kafka.consumer.KafkaStream;
import kafka.javaapi.consumer.ConsumerConnector;
import mockit.Injectable;
import mockit.Mocked;
import mockit.StrictExpectations;
import mockit.Tested;
import nl.haploid.octowight.TestData;
import org.junit.Test;

import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static org.junit.Assert.assertEquals;

public class KafkaConsumerFactoryTest {

	@Tested
	KafkaConsumerFactory service;

	@Injectable
	private ConsumerConfig consumerConfig;

	@Test
	@SuppressWarnings("unchecked")
	public void testCreateStream(final @Mocked ConsumerConnector kafkaConsumer,
	                             final @Mocked KafkaStream<byte[], byte[]> expectedStream) {
		final String topic = TestData.topic();
		final Map<String, List<KafkaStream<byte[], byte[]>>> streams = new HashMap<>();
		streams.put(topic, Collections.singletonList(expectedStream));
		new StrictExpectations() {{
			kafkaConsumer.createMessageStreams((Map<String, Integer>) any);
			times = 1;
			result = streams;
		}};
		final KafkaStream<byte[], byte[]> actualStream = service.createStream(kafkaConsumer, topic);
		assertEquals(expectedStream, actualStream);
	}
}

File: event-consumer/src/test/scala/nl/haploid/octowight/kafka/KafkaConsumerFactoryTest.scala
package nl.haploid.octowight.kafka

import java.util
import java.util.Collections

import _root_.kafka.consumer.{ConsumerConfig, KafkaStream}
import _root_.kafka.javaapi.consumer.ConsumerConnector
import nl.haploid.octowight._
import org.easymock.EasyMock

class KafkaConsumerFactoryTest extends AbstractTest {
  @Tested private[this] val kafkaConsumerFactory: KafkaConsumerFactory = null
  @Mocked private[this] val consumerConfig: ConsumerConfig = null

  "Kafka consumer factory" should "build a stream" in {
    val kafkaConsumer = mock[ConsumerConnector]
    val expectedStream = mock[KafkaStream[Array[Byte], Array[Byte]]]
    val topic = TestData.topic
    val streams = new util.HashMap[String, util.List[KafkaStream[Array[Byte], Array[Byte]]]]
    streams.put(topic, Collections.singletonList(expectedStream))
    expecting {
      kafkaConsumer.createMessageStreams(EasyMock.anyObject(classOf[util.Map[String, Integer]])) andReturn streams once()
    }
    whenExecuting(kafkaConsumer) {
      val actualStream = kafkaConsumerFactory.createStream(kafkaConsumer, topic)
      actualStream should be(expectedStream)
    }
  }
}
